L89-core-remove-grpc-insecure-channel-creation-api

```diff
@@ -0,0 +1,128 @@
+
+
+```
+/** Creates a secure channel using the passed-in credentials. Additional
+    channel level configuration MAY be provided by grpc_channel_args, though
+    the expectation is that most clients will want to simply pass NULL. The
+    user data in 'args' need only live through the invocation of this function.
+    However, if any args of the 'pointer' type are passed, then the referenced
+    vtable must be maintained by the caller until grpc_channel_destroy
+    terminates. See grpc_channel_args definition for more on this. */
+GRPCAPI grpc_channel* grpc_channel_create(const char* target,
+                                          grpc_channel_credentials* creds,
+                                          const grpc_channel_args* args);
+```
+
+Third, rename `grpc_server_add_secure_http2_port` to `grpc_server_add_http2_port` and move it from `grpc_security.h` to `grpc.h`.
+
+```
+/** Add a HTTP2 over an encrypted link over tcp listener.
+   Returns bound port number on success, 0 on failure.
+   REQUIRES: server not started */
+GRPCAPI int grpc_server_add_http2_port(grpc_server* server, const char* addr,
+                                       grpc_server_credentials* creds);
+```
+
+Fourth, add `grpc_channel_create_from_fd` and `grpc_server_add_channel_from_fd` to `grpc.h`. Note that the current APIs only support
+either insecure channel credentials (at client-side) or insecure server credentials (at server-side), and using other types of
+credentials will result in a failure. In future, the API's will be improved to support other types of credentials.
+
+
+```
+/** Create a secure channel to 'target' using file descriptor 'fd' and passed-in
+    credentials. The 'target' argument will be used to indicate the name for
+    this channel. Note that this API currently only supports insecure channel
+    credentials. Using other types of credentials will result in a failure. */
+GRPCAPI grpc_channel* grpc_channel_create_from_fd(
+    const char* target, int fd, grpc_channel_credentials* creds,
+    const grpc_channel_args* args);
+
+/** Add the connected secure communication channel based on file descriptor 'fd'
+   to the 'server' and server credentials 'creds'. The 'fd' must be an open file
+   descriptor corresponding to a connected socket. Events from the file
+   descriptor may come on any of the server completion queues (i.e completion
+   queues registered via the grpc_server_register_completion_queue API).
+   The 'reserved' pointer MUST be NULL.
+   TODO(hork): add channel_args to this API to allow endpoints and transports
+   created in this function to participate in the resource quota feature.
+   Note that this API currently only supports inseure server credentials
+   Using other types of credentials will result in a failure. */
+GRPCAPI void grpc_server_add_channel_from_fd(grpc_server* server,
+                                             void* reserved, int fd,
```

Comment:
Let's get rid of this `reserved` parameter while we're changing this. I don't think we need it.

Reply:
Done.